File: testing/Lab2/src/main/scala/function/Main.scala
package function

object Main {
  def main(args: Array[String]): Unit = {
    val sine = new Sinus
    val log = new Log
    val trigEvaluator = new TrigEvaluator(sine)
    val logEvaluator = new LogEvaluator(log)
    val funcEvaluator = new FuncEvaluator(trigEvaluator, logEvaluator)
    funcEvaluator.writeCsvFile("utils/res.csv", -Math.PI * 3, 0.1, 0)
  }
}

File: testing/Lab2/src/main/kotlin/function/Main.kt
package function

fun main() {
    val sine = Sinus()
    val log = Log()
    val trigEvaluator = TrigEvaluator(sine)
    val logEvaluator = LogEvaluator(log)
    val funcEvaluator = FuncEvaluator(trigEvaluator, logEvaluator)

    print("Enter csv file path: ")
    val filepath = readLine()!!
    print("Enter start of the interval: ")
    val startX = readLine()!!.toDouble()
    print("Enter end of the interval: ")
    val endX = readLine()!!.toDouble()
    print("Enter step: ")
    val step = readLine()!!.toDouble()

    funcEvaluator.writeCsvFile(filepath, startX, step, endX)

    println("The data successfully written to $filepath")
}